node/src/database/rocksdb.rs

```diff
@@ -268,13 +292,33 @@ impl<'db, DB: DBAccess> Ledger for DBTransaction<'db, DB> {
     }
 
     fn fetch_block(&self, hash: &[u8]) -> Result<Option<ledger::Block>> {
-        let blk = self
-            .snapshot
-            .get_cf(self.ledger_cf, hash)?
-            .map(|blob| ledger::Block::read(&mut &blob[..]))
-            .transpose()?;
+        match self.snapshot.get_cf(self.ledger_cf, hash)? {
+            Some(blob) => {
+                let record = HeaderRecord::read(&mut &blob[..])?;
+
+                // Retrieve all transactions buffers with single call
+                let txs_buffers = self.snapshot.multi_get_cf(
+                    record
+                        .transactions_ids
+                        .iter()
+                        .map(|id| (self.ledger_txs_cf, id))
+                        .collect::<Vec<(&ColumnFamily, &[u8; 32])>>(),
+                );
+
+                let mut txs = vec![];
+                for buf in txs_buffers {
+                    let mut buf = buf?.unwrap();
```

Comment:
Is unwrap here ok? Also other instances in case they panic

Reply:
This is unrecoverable error. If we encode/decode block transactions partially, we should not continue.